src/tpg_port.c

```diff
@@ -515,32 +500,37 @@ static bool port_setup_port(uint8_t port)
             .mq_mode        = ETH_MQ_RX_RSS,
             .max_rx_pkt_len = port_dev_info[port].pi_dev_info.max_rx_pktlen,
             .split_hdr_size = 0,
-            .header_split   = 0, /**< Header Split disabled */
-            .hw_ip_checksum = 1, /**< IP checksum offload enabled */
-            .hw_vlan_filter = 0, /**< VLAN filtering disabled */
-            .jumbo_frame    = 1, /**< Jumbo Frame Support enabled */
-            .hw_strip_crc   = 0, /**< CRC stripped by hardware */
-            .enable_scatter = 1,
+            .offloads       = DEV_RX_OFFLOAD_IPV4_CKSUM |
+                              DEV_RX_OFFLOAD_JUMBO_FRAME |
+                              DEV_RX_OFFLOAD_SCATTER |
+                              DEV_RX_OFFLOAD_KEEP_CRC,
         },
         .rx_adv_conf = {
             .rss_conf = {
                 .rss_key = port_rss_key,
                 .rss_key_len = sizeof(port_rss_key),
-                .rss_hf = ETH_RSS_IPV4 | ETH_RSS_NONFRAG_IPV4_TCP | ETH_RSS_NONFRAG_IPV4_UDP,
+                .rss_hf = ETH_RSS_IPV4 |
+                          ETH_RSS_NONFRAG_IPV4_TCP |
+                          ETH_RSS_NONFRAG_IPV4_UDP,
             },
         },
         .txmode = {
+#if !defined(TPG_SW_CHECKSUMMING)
```

Comment:
@insanum Sorry for taking so long in reviewing this PR. I finally got the chance to test it out and everything works fine except for when we compile with `TPG_SW_CHECKSUMMING` and the NIC supports TX offloading. In that case we send the mbufs with TX offload configured but the NIC is not configured for TX offload because of this `#if` here.

If I remove it then everything works fine in all cases.

I can fix it up by adding a new commit to your branch if that's ok with you. Also, do you mind if I add you to the CONTRIBUTORS file using the public information on your github profile?

Thanks,
Dumitru